### PR TITLE
Create separate `documentation.yaml` file

### PR DESCRIPTION
**Pull request by RedbackThomson**
Closes https://github.com/aws-controllers-k8s/community/issues/1223
Expands upon https://github.com/aws-controllers-k8s/code-generator/pull/394

Description of changes:
This pull request moves the custom GoDoc overrides from inside the `generator.yaml` file to a new, separate `documentation.yaml` file. This `documentation.yaml` file currently supports modify field GoDocs (but could be extended to modify resource GoDocs as well) in the following ways:
- Prepend: Add a comment before the existing shape ref GoDoc
- Append: Add a comment after the existing shape ref GoDoc
- Override: Completely replace the existing shape ref GoDoc

The current RDS documentation modifications are now modeled in the `documentation.yaml` like so:
```yaml
resources:
  DBClusterParameterGroup:
    fields:
      Parameters: 
        override: |
          DEPRECATED - do not use.  Prefer ParameterOverrides instead.
      ParameterOverrides: 
        append: |
          These are ONLY user-defined parameter overrides for the
          DB cluster parameter group. This does not contain default or system
          parameters.
  DBParameterGroup:
    fields:
      ParameterOverrides: 
        append: |
          These are ONLY user-defined parameter overrides for the DB parameter
          group. This does not contain default or system parameters.
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/model/field.go [hunk 1]
```diff
@@ -88,36 +89,76 @@ type Field struct {
...
+// split it by lines, strip it of leading whitespace and slashes and prepend
+// `//` to each line.
+func (f *Field) formatUserProvidedDocstring(in string) string {
+	var sb strings.Builder
+	// Strip any leading comment slashes from the config option
+	// docstring since we'll be automatically adding the Go comment
+	// slashes to the beginning of each new line
+	lines := strings.Split(in, "\n")
+
+	// Traverse in reverse order until, deleting empty lines
```
Comment (jaypipes) [inline pkg/model/field.go:148]: Any reason you wanted to delete empty newlines? Sometimes a long docstring is better broken into paragraphs.

**Review by jaypipes — COMMENTED** [review]
great stuff here @RedbackThomson! really appreciate this. Couple smallish nits inline but overall awesome.

**Comment by jaypipes** [comment]
@RedbackThomson also, see unit test output failure:

```
 	want (*model.SDKAPI, string, string, "github.com/aws-controllers-k8s/code-generator/pkg/config".Config, "github.com/aws-controllers-k8s/code-generator/pkg/config".DocumentationConfig)
# github.com/aws-controllers-k8s/code-generator/pkg/testutil [github.com/aws-controllers-k8s/code-generator/pkg/generate/ack.test]
pkg/testutil/schema_helper.go:91:73: not enough arguments in call to ackmodel.New
	have (*model.SDKAPI, string, string, "github.com/aws-controllers-k8s/code-generator/pkg/config".Config)
	want (*model.SDKAPI, string, string, "github.com/aws-controllers-k8s/code-generator/pkg/config".Config, "github.com/aws-controllers-k8s/code-generator/pkg/config".DocumentationConfig)
FAIL	github.com/aws-controllers-k8s/code-generator/pkg/generate/ack [build failed]
FAIL	github.com/aws-controllers-k8s/code-generator/pkg/generate/code [build failed]
?   	github.com/aws-controllers-k8s/code-generator/pkg/generate/crossplane	[no test files]
?   	github.com/aws-controllers-k8s/code-generator/pkg/generate/olm	[no test files]
?   	github.com/aws-controllers-k8s/code-generator/pkg/generate/templateset	[no test files]
?   	github.com/aws-controllers-k8s/code-generator/pkg/metadata	[no test files]
FAIL	github.com/aws-controllers-k8s/code-generator/pkg/model [build failed]
FAIL	github.com/aws-controllers-k8s/code-generator/pkg/model/multiversion [build failed]
ok  	github.com/aws-controllers-k8s/code-generator/pkg/sdk	0.319s
make: *** [Makefile:56: test] Error 2 
```

File: pkg/model/field.go (hunk 1 above)
Comment (RedbackThomson) [reply pkg/model/field.go:148]: Only deleting trailing newlines here

**Comment by RedbackThomson** [comment]
@A-Hilaly @jaypipes Unit tests passing + added code doc

**Comment by A-Hilaly** [comment]
Nice, ty @RedbackThomson !
/lgtm

**Comment by RedbackThomson** [comment]
This will require a follow up with the RDS controller to move their stuff over. Reminder to me to do that

**Pull request merged**